odoo/models.py

```diff
@@ -4707,6 +4774,19 @@ def _write_multi(self, vals_list):
                         column=column,
                         expr=expr,
                     )
+                if field.company_dependent:
+                    fallback = self.env['ir.default']._get_model_defaults(self._name).get(field.name)
```

Comment:
What about using the field method instead?
```suggestion
                    fallback = field.get_company_dependent_fallback(self)
```